assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -0,0 +1,679 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:EditBuildConfigController
+ * @description
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('EditBuildConfigController', function ($scope, $routeParams, DataService, ProjectsService, BuildConfigsService, $filter, ApplicationGenerator, Navigate, $location, AlertMessageService, SOURCE_URL_PATTERN) {
+
+    $scope.projectName = $routeParams.project;
+    $scope.buildConfig = null;
+    $scope.alerts = {};
+    $scope.emptyMessage = "Loading...";
+    $scope.sourceURLPattern = SOURCE_URL_PATTERN;
+    $scope.options = {};
+    $scope.openshiftBuilderImages = ["ruby","python","perl","php","nodejs","wildfly"]
```

Comment:
They should be in `spec.tags.tags` I thought? We use the builder tag in the add to project catalog.

I'd rather show all than have a magic list. This is a problem if there is a valid builder we're not showing. What happens if I edit a build config that uses an image not in this array? We need to handle that case even if we update to use the builder tag.